File: src/main/java/mages/grimoire/model/Spell.java
package mages.grimoire.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;

/** Spell */
@Data
public class Spell {

  @Id private int spellId;
  private String spellName;
  private int level;
  private boolean ritual;
  private String castingTime;

  @JsonProperty("range")
  private int rangeValue;

  private boolean verbal;
  private boolean somatic;
  private String material;
  private String school;
  private String duration;
  private boolean concentration;
  private String target;
  private String saving_throw;
  private String description;
  private String higherLevels;
  private String damage;
  private String damageType;
  @Transient private List<ClassType> classType;
  private String book;

  public static enum School {
    Abjuration,
    Conjuration,
    Divination,
    Enchantment,
    Evocation,
    Illusion,
    Necromancy,
    Transmutation;

  }

  public static enum Target {
    Self,
    Creature,
    Cone,
    Cube,
    Cylinder,
    Line,
    Sphere
  }

  public static enum DamageType {
    Acid,
    Bludgeoning,
    Cold,
    Fire,
    Force,
    Lightning,
    Necrotic,
    Piercing,
    Poison,
    Psychic,
    Radiant,
    Slashing,
    Thunder
  }
}

File: src/main/angular/src/app/shared/interfaces/spell.ts
export interface Spell {

  spellId?: number;
  spellName: string;
  level: number;
  ritual: boolean;
  castingTime: string;

  rangeValue: number;

  verbal: boolean;
  somatic: boolean;
  material: string;
  school: School;
  duration: number;
  concentration: boolean;
  target: Target;
  saving_throw: string;
  description: string;
  higherLevels: string;
  damage: string;
  damageType: DamageType;
  classType: ClassType;
  book: string;
}

export enum School {
  Abjuration = 'Abjuration',
  Conjuration = 'Conjuration',
  Divination = 'Divination',
  Enchantment = 'Enchantment',
  Evocation = 'Evocation',
  Illusion = 'Illusion',
  Necromancy = 'Necromancy',
  Transmutation = 'Transmutation'
}

export enum Target {
  Self = 'Self',
  Creature = 'Creature',
  Cone = 'Cone',
  Cube = 'Cube',
  Cylinder = 'Cylinder',
  Line = 'Line',
  Sphere = 'Sphere'
}

export enum DamageType {
  Acid = 'Acid',
  Bludgeoning = 'Bludgeoning',
  Cold = 'Cold',
  Fire = 'Fire',
  Force = 'Force',
  Lightning = 'Lightning',
  Necrotic = 'Necrotic',
  Piercing = 'Piercing',
  Poison = 'Poison',
  Psychic = 'Psychic',
  Radiant = 'Radiant',
  Slashing = 'Slashing',
  Thunder = 'Thunder'
}

export enum ClassType {
  Bard = 'Bard',
  Cleric = 'Cleric',
  Druid = 'Druid',
  Paladin = 'Paladin',
  Ranger = 'Ranger',
  Sorcerer = 'Sorcerer',
  Warlock = 'Warlock',
  Wizard = 'Wizard'
}
